src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/AllocateYFlowResourcesAction.java

```diff
@@ -157,9 +160,21 @@ public void perform(S from, S to, E event, C context, T stateMachine) {
                     yFlowToUpdate.setProtectedPathMeterId(null);
                 }
                 yFlowToUpdate.setSharedEndpointMeterId(newResources.getSharedEndpointResources().getMeterId());
+
+                List<SubFlowPathInfo> subPathList = new ArrayList<>();
+                for (YSubFlow subFlow : yFlow.getSubFlows()) {
+                    Flow flow = subFlow.getFlow();
+                    for (FlowPath path : flow.getPaths()) {
+                        subPathList.add(new SubFlowPathInfo(
+                                flow.getFlowId(), path.getCookie(), path.getMeterId(),
+                                FlowPathMapper.INSTANCE.mapToPathNodes(flow, path),
+                                flow.getVlanStatistics(), path.hasIngressMirror(), path.hasEgressMirror()));
+                    }
+                }
+                return subPathList;
```

Comment:
[minor] I'm wondering if here we can use one of those fancy streams to somehow collect the subPathList. But I don't really know how to do that or if it possible.